### PR TITLE
feat: use moov-io/base logger instead of stdlib

### PR DESCRIPTION
**Pull request by adamdecaf**
It's bad design for libraries to log and breaks convention by not using moov's logger interface.

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -49,6 +52,7 @@ type Option func(*Options) error
 
 func GetDefaultOptions() Options {
 	return Options{
+		Logger:      log.NewNopLogger(),
```
Comment (atonks2) [inline options.go:55]: Would `log.NewDefaultLogger()` be applicable to more use-cases?

**Comment by alovak** [comment]
I wanted to add error handler so developers could use their own logs or whatever

**Comment by alovak** [comment]
this aspec (logging) the same as error handling has never been addressed.

File: options.go (hunk 1 above)
Comment (adamdecaf) [reply options.go:55]: Generally libraries shouldn't log. It's "bad form."

**Comment by adamdecaf** [comment]
@alovak error handling would be good to include at some point. I just noticed these stdlib log calls in some testing.

File: options.go (hunk 1 above)
Comment (atonks2) [reply options.go:55]: Oh I see what you mean. I thought you were only saying it was bad to not use Go's standard logger interface. Yeah that makes sense.

**Review by atonks2 — APPROVED** [review]

**Comment by alovak** [comment]
@adamdecaf yeah, I just had no other way at that time to see it.

Let's maybe add `errorHandler` instead of `logger`. I can do it by the beginning of the next week. If it will be obvious like `s.Opts.ErrorHandler(err)`?

**Comment by adamdecaf** [comment]
That works. I'm going to close this PR and we can remove the stdlib logging with the error handler.

**Comment by alovak** [comment]
@adamdecaf added error handler in #14

**Pull request closed without merging**